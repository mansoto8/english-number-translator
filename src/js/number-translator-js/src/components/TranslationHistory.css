#history-button {
    display: block;
    margin-top: 40px;
    margin-bottom: 15px;
}

#history-table {
    font-family: Arial, Helvetica, sans-serif;
    border-collapse: collapse;
    width: 100%;
}

#history-table td, #customers th {
    border: 1px solid #ddd;
    padding: 8px;
}

#history-table tr:nth-child(even){background-color: #f2f2f2;}

#history-table tr:hover {background-color: #ddd;}

#history-table th {
    padding-top: 12px;
    padding-bottom: 12px;
    text-align: center;
    background-color: #545b62;
    color: white;
}
