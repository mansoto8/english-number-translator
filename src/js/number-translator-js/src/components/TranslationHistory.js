import React, {Component} from 'react';
import {Button, Form} from 'reactstrap';

class TranslationHistory extends Component {

    translations = [];

    constructor(props) {
        super(props);
        this.state = {
            translations: this.translations,
        };
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    async handleSubmit(event) {
        event.preventDefault();

        const response = await fetch('/api/numbers', {
            method: 'GET',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            },
        });

        const translations = await response.json();
        this.setState({translations});
    }

    render() {
        const {translations} = this.state;

        return <div>
            <h2>Translation list</h2>
            <div>
                <Form onSubmit={this.handleSubmit}>
                    <Button color="primary" type="submit">Check Translation History</Button>{' '}
                </Form>
                {translations.length == 0 ? (<p>No registers</p>) : (<p></p>)}
                <table>
                    <thead>
                        <tr>
                            <th>Number</th>
                            <th>Translation</th>
                        </tr>
                    </thead>
                    <tbody>
                        {translations.map(translation =>
                            <tr key={translation.number}>
                                <td>{translation.number}</td>
                                <td>{translation.translation}</td>
                            </tr>
                        )}
                    </tbody>
                </table>
            </div>
        </div>
    }
}

export default TranslationHistory;
