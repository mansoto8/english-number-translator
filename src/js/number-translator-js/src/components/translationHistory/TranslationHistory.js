import React, {Component} from 'react';
import {Container,Button} from 'reactstrap';
import './TranslationHistory.css';

class TranslationHistory extends Component {

    translations = [];

    constructor(props) {
        super(props);
        this.state = {
            translations: this.translations,
        };
        this.getHistory = this.getHistory.bind(this);
    }

    async getHistory(event) {
        event.preventDefault();

        const response = await fetch('/api/numbers', {
            method: 'GET',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            },
        });

        const translations = await response.json();
        this.setState({translations});
    }

    render() {
        const {translations} = this.state;

        return <div>
            <Container>
                <Button id="history-button" color="primary" type="button"
                        onClick={this.getHistory}>Check Translation History</Button>
                <table id="history-table">
                    <thead>
                    <tr>
                        <th>Number</th>
                        <th>Translation</th>
                    </tr>
                    </thead>
                    <tbody>
                    {translations.map(translation =>
                        <tr key={translation.number}>
                            <td>{translation.number.toLocaleString()}</td>
                            <td>{translation.translation}</td>
                        </tr>
                    )}
                    </tbody>
                </table>
                {translations.length === 0 ? (<p>No registers</p>) : (<p></p>)}
            </Container>
        </div>
    }
}

export default TranslationHistory;
