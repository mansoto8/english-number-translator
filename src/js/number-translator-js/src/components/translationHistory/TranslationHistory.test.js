import TranslationHistory from './TranslationHistory';
import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import {configure} from 'enzyme';
import {shallow} from 'enzyme';

configure({adapter: new Adapter()});

it('renders Translation history without registers', () => {
    const tree = shallow(
        <TranslationHistory/>,
    );

    expect(tree).toMatchSnapshot();
});

it('renders Translation history with register', (done) => {
    const mockSuccessResponse = [
        {
            "number": -1,
            "translation": "Negative one"
        },
        {
            "number": 20,
            "translation": "Twenty"
        },
        {
            "number": 999,
            "translation": "Nine hundred and ninety nine"
        }
    ];

    const mockJsonResponse = Promise.resolve(mockSuccessResponse);
    const mockFetchPromise = Promise.resolve({
        json: () => mockJsonResponse,
    });
    jest.spyOn(global, 'fetch').mockImplementation(() => mockFetchPromise);

    const wrapper = shallow(
        <TranslationHistory/>
    );

    const button = wrapper.find('#history-button');
    button.simulate('click', {
        preventDefault() {
        }
    });

    process.nextTick(() => {
        expect(wrapper).toMatchSnapshot();
        done();
    });
});
