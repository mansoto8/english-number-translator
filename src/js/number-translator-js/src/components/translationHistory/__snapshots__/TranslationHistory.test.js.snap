// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Translation history with register 1`] = `
<div>
  <Container
    tag="div"
  >
    <Button
      color="primary"
      id="history-button"
      onClick={[Function]}
      tag="button"
      type="button"
    >
      Check Translation History
    </Button>
    <table
      id="history-table"
    >
      <thead>
        <tr>
          <th>
            Number
          </th>
          <th>
            Translation
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          key="-1"
        >
          <td>
            -1
          </td>
          <td>
            Negative one
          </td>
        </tr>
        <tr
          key="20"
        >
          <td>
            20
          </td>
          <td>
            Twenty
          </td>
        </tr>
        <tr
          key="999"
        >
          <td>
            999
          </td>
          <td>
            Nine hundred and ninety nine
          </td>
        </tr>
      </tbody>
    </table>
    <p />
  </Container>
</div>
`;

exports[`renders Translation history without registers 1`] = `
<div>
  <Container
    tag="div"
  >
    <Button
      color="primary"
      id="history-button"
      onClick={[Function]}
      tag="button"
      type="button"
    >
      Check Translation History
    </Button>
    <table
      id="history-table"
    >
      <thead>
        <tr>
          <th>
            Number
          </th>
          <th>
            Translation
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
    <p>
      No registers
    </p>
  </Container>
</div>
`;
